log/logtest/recorder.go

```diff
@@ -141,22 +118,55 @@ func (r *Recorder) Result() []*ScopeRecords {
 	defer r.mu.Unlock()
 
 	ret := []*ScopeRecords{}
-	ret = append(ret, r.currentScopeRecord)
 	for _, l := range r.loggers {
-		ret = append(ret, l.Result()...)
+		ret = append(ret, l.scopeRecord)
 	}
 	return ret
 }
 
-// Reset clears the in-memory log records.
+// Reset clears the in-memory log records for all loggers.
 func (r *Recorder) Reset() {
 	r.mu.Lock()
 	defer r.mu.Unlock()
 
-	if r.currentScopeRecord != nil {
-		r.currentScopeRecord.Records = nil
-	}
 	for _, l := range r.loggers {
 		l.Reset()
 	}
 }
+
+type logger struct {
+	embeddedLogger // nolint:unused  // Used to embed embedded.Logger.
```

Comment:
I think now we can remove `embeddedLogger` and do simply

```suggestion
	embedded.Logger
```